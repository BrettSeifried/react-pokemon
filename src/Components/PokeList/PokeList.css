.table {
  display: flex;
  flex-wrap: wrap;
}

.table {
  display: grid;
  grid-template-columns: 1fr 1fr 1fr 1fr;
  grid-template-rows: 1fr 1fr 1fr 1fr 1fr;
  gap: 1em 1em;
  grid-template-areas:
    '. . . .'
    '. . . .'
    '. . . .'
    '. . . .'
    '. . . .';
}

.pokeCard {
  display: flex;
  flex-direction: column;
  align-items: center;
  /* border: black 2px solid; */
  flex-wrap: wrap;
  border-radius: 5%;
  background-image: url('../../card.jpg');
  background-size: contain;
  background-repeat: no-repeat;
  height: 350px;
  width: 100%;
}

img {
  height: 100px;
  margin-top: 7%;
  margin-bottom: 13%;
}

.nameUpper {
  text-transform: uppercase;
  margin-top: 3%;
}

span {
  padding: 3%;
}
