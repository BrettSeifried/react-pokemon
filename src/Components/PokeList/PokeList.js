import React from 'react';
import './PokeList.css';
import { Button } from '@mui/material';

export default function PokeList({ pokemon, setLoading, setPage }) {
  const handleClick = () => {
    setLoading(true);
    setPage((prevState) => ++prevState);
  };

  return (
    <div>
      {/* <button className="nextPage" onClick={handleClick}>
        Next
      </button> */}
      <div className="table">
        {pokemon.map((poke) => (
          <p key={poke.id} className="pokeCard">
            <img alt={poke.id} src={poke.url_image} />
            <span className="nameUpper">{poke.pokemon}</span>
            <span>HP: {poke.hp}</span>
            <span>Attack: {poke.attack}</span>
            <span>Defense: {poke.attack}</span>
            <span>Type: {poke.type_1}</span>
          </p>
        ))}
      </div>
      <Button
        style={{ backgroundColor: 'teal', color: 'white' }}
        className="nextPage"
        onClick={handleClick}
      >
        Next Page
      </Button>
    </div>
  );
}

{
  /* {pokemon.map((poke) => (
  <p key={poke.id}>
    {poke.pokemon}
    <img alt={poke.pokemon} src={poke.url_image}></img>
  </p>
))}
; */
}
