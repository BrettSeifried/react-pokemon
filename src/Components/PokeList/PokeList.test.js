import { render } from '@testing-library/react';
import PokeList from './PokeList';

test('Renders Pokemon Name', () => {
  const pokeName = render(<PokeList pokemon={['pikachu']} />);
  expect(pokeName).toMatchSnapshot();
});
