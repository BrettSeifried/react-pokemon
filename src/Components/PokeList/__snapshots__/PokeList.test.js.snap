// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders Pokemon Name 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <div
          class="table"
        >
          <p
            class="pokeCard"
          >
            <img />
            <span
              class="nameUpper"
            />
            <span>
              HP: 
            </span>
            <span>
              Attack: 
            </span>
            <span>
              Defense: 
            </span>
            <span>
              Type: 
            </span>
          </p>
        </div>
        <button
          class="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButtonBase-root nextPage css-1e6y48t-MuiButtonBase-root-MuiButton-root"
          style="background-color: teal; color: white;"
          tabindex="0"
          type="button"
        >
          Next Page
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <div
        class="table"
      >
        <p
          class="pokeCard"
        >
          <img />
          <span
            class="nameUpper"
          />
          <span>
            HP: 
          </span>
          <span>
            Attack: 
          </span>
          <span>
            Defense: 
          </span>
          <span>
            Type: 
          </span>
        </p>
      </div>
      <button
        class="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButtonBase-root nextPage css-1e6y48t-MuiButtonBase-root-MuiButton-root"
        style="background-color: teal; color: white;"
        tabindex="0"
        type="button"
      >
        Next Page
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
